drivers/adc/adc_nrfx_saadc.c

```diff
@@ -49,16 +62,43 @@ BUILD_ASSERT((NRF_SAADC_AIN0 == NRF_SAADC_INPUT_AIN0) &&
 	     "Definitions from nrf-adc.h do not match those from nrf_saadc.h");
 #endif
 
+#ifdef CONFIG_SOC_NRF54H20
+
+/* Haltium always uses bounce buffers in RAM */
+
+#define SAADC_MEMORY_SECTION					                     \
+	COND_CODE_1(DT_NODE_HAS_PROP(DT_NODELABEL(adc), memory_regions), \
+		(__attribute__((__section__(LINKER_DT_NODE_REGION_NAME(	     \
+			DT_PHANDLE(DT_NODELABEL(adc), memory_regions)))))),	     \
+		())
+
+#define SAADC_BUFFER_SIZE SAADC_CH_NUM
+
+static uint8_t adc_samples_buffer[SAADC_BUFFER_SIZE] SAADC_MEMORY_SECTION;
+
+#define ADC_BUFFER_IN_RAM 1
+
+#endif /* CONFIG_SOC_NRF54H20 */
+
 struct driver_data {
 	struct adc_context ctx;
 
 	uint8_t positive_inputs[SAADC_CH_NUM];
+
+#if ADC_BUFFER_IN_RAM
```

Comment:
```suggestion
#ifdef ADC_BUFFER_IN_RAM
```

Reply:
I will change 
`#define ADC_BUFFER_IN_RAM 1`  few lines earlier to:
`#define ADC_BUFFER_IN_RAM`
and 
`#if ADC_BUFFER_IN_RAM`
to
`#if defined(ADC_BUFFER_IN_RAM)`